.travis.yml

```diff
@@ -1,9 +1,34 @@
 git:
   quiet: true
 
-dist: trusty
+matrix:
+  include:
+    - os: linux
+      dist: trusty
+      compiler: gcc
+      env: ENABLE_MAINTAINER_ZTS=0 ENABLE_DEBUG=0 ENABLE_OPCACHE=1
+    - os: linux
+      dist: trusty
+      compiler: gcc
+      env: ENABLE_MAINTAINER_ZTS=0 ENABLE_DEBUG=0 ENABLE_OPCACHE=0
+    - os: linux
+      dist: trusty
+      compiler: gcc
+      env: ENABLE_MAINTAINER_ZTS=1 ENABLE_DEBUG=1 ENABLE_OPCACHE=0
+    - os: osx
+      osx_image: xcode9.4
+      compiler: clang
+      env: ENABLE_MAINTAINER_ZTS=0 ENABLE_DEBUG=0 ENABLE_OPCACHE=1
+    - os: osx
+      osx_image: xcode9.4
+      compiler: clang
+      env: ENABLE_MAINTAINER_ZTS=1 ENABLE_DEBUG=1 ENABLE_OPCACHE=0
```

Comment:
I think this is too many jobs, and will result in long queue times. I'd suggest to keep only the existing two linux jobs, and add one osx job. The osx job can use a zts/debug/opcache combination that is not used in the linux jobs.

Reply:
During my tests, while running these 5 jobs they didn't really get queued. This will overall make the feedback a bit slower as it seems to also wait for the osx build before reporting back to GH. (I was hoping it would give the feedback as soon as all the non-"allow_failures" were done). I'd suggest we try this matrix first and then reduce it if it becomes a problem.

Reply:
To get the feedback early as you hoped add [`fast_finish: true`](https://docs.travis-ci.com/user/customizing-the-build/#fast-finishing) to the `matrix` section.

Reply:
You're not going to see much queuing with one person pushing to one branch. The problem is that it's common for us to have, say, someone push four branches (changes landed on 7.2 and merged up), and then maybe also have some PRs running. We already see queuing with just the two jobs we have right now. It's not very problematic because our (Linux) build times are relatively low, so even with some queuing you tend to get results in a timely manner. Going from two to five builds will make this a lot more problematic, so I think we should really pick which builds are worthwhile and which aren't. A new osx build is very valuable, but another Linux builder with slightly different configuration is not.

Reply:
To be more precise, it looks like we have 5 parallel jobs on Travis. This means that right now we can run 2.5 builds in parallel. With 3 jobs we can run 1.7 builds in parallel. With 5 jobs we can run only one build.